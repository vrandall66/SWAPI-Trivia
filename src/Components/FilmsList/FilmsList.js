import React from 'react';
import { Link } from 'react-router-dom';
import './FilmsList.css'

const FilmsList = ({films}) => {
  const filmsToLi = films.map( film => <li className='filmLi'>{film}</li>)
  
  return (
    <ul className='filmUl'>
      <li className='filmLi'>Also Seen In:</li>
      {filmsToLi}
    </ul>
  )
}

export default FilmsList