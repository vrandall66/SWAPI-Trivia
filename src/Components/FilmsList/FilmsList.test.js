import React from "react";
import { shallow } from "enzyme";
import FilmsList from "./FilmsList";

describe("FilmsList", () => {
  let wrapper;
  let films = [
    "The Empire Strikes Back",
    "Attack of the Clones",
    "The Phantom Menace",
    "Revenge of the Sith",
    "Return of the Jedi",
    "A New Hope",
    "The Force Awakens"
  ];

  beforeEach(() => {
    wrapper = shallow(<FilmsList films={films} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
