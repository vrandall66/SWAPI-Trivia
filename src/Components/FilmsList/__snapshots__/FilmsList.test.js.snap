// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList should match the snapshot 1`] = `
<ul
  className="filmUl"
>
  <li
    className="filmLi"
  >
    Also Seen In:
  </li>
  <li
    className="filmLi"
    key="0"
  >
    The Empire Strikes Back
  </li>
  <li
    className="filmLi"
    key="1"
  >
    Attack of the Clones
  </li>
  <li
    className="filmLi"
    key="2"
  >
    The Phantom Menace
  </li>
  <li
    className="filmLi"
    key="3"
  >
    Revenge of the Sith
  </li>
  <li
    className="filmLi"
    key="4"
  >
    Return of the Jedi
  </li>
  <li
    className="filmLi"
    key="5"
  >
    A New Hope
  </li>
  <li
    className="filmLi"
    key="6"
  >
    The Force Awakens
  </li>
</ul>
`;
